hydra/test_utils/launcher_common_tests.py

```diff
@@ -22,14 +22,18 @@ class LauncherTestSuite:
     def task_function(self, cfg: DictConfig) -> Any:
         return 100
 
+    def use_task_function(self) -> bool:
+        return True
+
     def test_sweep_1_job(
-        self, sweep_runner: TSweepRunner, launcher_name: str, overrides: List[str],
+        self, sweep_runner: TSweepRunner, launcher_name: str, overrides: List[str]
     ) -> None:
+
         sweep_1_job(
             sweep_runner,
             overrides=["hydra/launcher=" + launcher_name, "hydra.sweep.dir=."]
             + overrides,
-            task_function=self.task_function,
+            task_function=self.task_function if self.use_task_function() else None,
```

Comment:
instead of having a special use_task_function(), can you just have a None task_function?

Reply:
yes, I don't remember why i did this 😵
Ive removed the function in the next commit.

Reply:
So one cleaner way I thought of is to do something like 
```python

class RayLauncherTestSuite(LauncherTestSuite):
    """
        Run the Launcher test suite on this launcher.
        """

    def __init__(self):
        self.task_function = None


@pytest.mark.skipif(sys.platform.startswith("win"), reason=win_msg)
@pytest.mark.parametrize("launcher_name, overrides", [("ray_local", [])])
class TestRayLocalLauncher(RayLauncherTestSuite):
    """
    Run the Launcher test suite on this launcher.
    """
    pass
```

But having a ```__init__``` yields this warning

```python

============================================================== warnings summary ===============================================================
plugins/hydra_ray_launcher/tests/test_ray_local_launcher.py:40
  /Users/jieru/workspace/hydra-fork/hydra/plugins/hydra_ray_launcher/tests/test_ray_local_launcher.py:40: PytestCollectionWarning: cannot collect test class 'TestRayLocalLauncher' because it has a __init__ constructor (from: plugins/hydra_ray_launcher/tests/test_ray_local_launcher.py)
    class TestRayLocalLauncher(RayLauncherTestSuite):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================================= 12 passed, 1 warning in 38.30s ========================================================
(pytest38) jieru-mbp:hydra jieru$ 

```

I haven't been able to figure out a cleaner way to do this :(

Reply:
Actually I thought of somethign that might work... trying it out now.